### DISCUSSION TITLE
K22F updates using DFU and MCUXpresso

### DISCUSSION THREAD

**Discussion by karamanr** [zephyrproject-rtos/zephyr#39360] [Q&A]
Hello,

The main idea of my project is to update the software OTA using MCUBoot, DFU and Mender. I followed this https://hub.mender.io/t/updating-device-firmware-using-dfu-in-zephyr-project-on-a-frdm-k64f-board/1618 to set up MCUBoot and DFU for my board and it all works fine. However, the next step is putting and application build with MCUXpresso on the K22F board. What is the best way to approach this? Do I need to use tools from the Zephyr Project to build my application or is there a tool that I can use with MCUXpresso to build and sign my application? I have tried using the binary built by MCUXpresso and sign it with the Zephyr tools, however it seems that MCUBoot is unable to detect the image when I flash it to the board using DFU.

**Comment by DerekSnell** [comment]
Hi Karamanr,

As you found in the Mender tutorial, Zephyr Project makes it pretty easy to build a Zephyr application intended to use with MCUboot.  There are some requirements for working with MCUboot, like partitioning the flash for the bootloader and the application image slots. 
 Plus MCUboot uses an image format that the bootloader needs to load the images.  See https://www.mcuboot.com/documentation/design/#boot-loader for more details from MCUboot.  With Zephyr, when you build the application with CONFIG_BOOTLOADER_MCUBOOT=y, much of these steps are done for you behind the scenes.

If you want to use this open-source MCUboot project and don't want to use Zephyr in your application, then you will need to modify your MCUXpresso application project to build the image for the proper slot location, add the image header, and maybe other details to be compatible with the MCUboot bootloader, following the documentation at the above link.  MCUboot includes the imgtool for signing, so once the application is built, you can use it to sign it.

Or your other option is to join us here at the Zephyr Project.  Use Zephyr in your application, and use these tools to easily build and sign your app for MCUboot.  To get started in general with Zephyr, check out [Getting Started Guide](https://docs.zephyrproject.org/latest/getting_started/index.html).  And each board has details to get started with it, see [FRDM-K64F](https://docs.zephyrproject.org/2.6.0/boards/arm/frdm_k64f/doc/index.html) or [FRDM-K22F](https://docs.zephyrproject.org/latest/boards/arm/frdm_k22f/doc/index.html).

Best regards
